File: word2json/models/media.py
# %%
from typing import Optional
from word2json.models.base import *


class Media(BaseDocument):
    title: Optional[str]
    subtitle: Optional[str]
    slug: Optional[Slug]
    author: Optional[str]
    mediaUrl: Optional[str]
    category: Optional[str]
    tags: Optional[list[str]]
    mainImageUrl: Optional[str]
    description: Optional[str]
    body: Optional[list[Body]]
    writtenAt: Optional[str]
    publishedAt: Optional[str]
    downloadLink: Optional[str]
    downloadMeta: Optional[str]

File: json2word/models/media.ts
import { Body, Slug, BaseDocument } from "./base";

export interface Media extends BaseDocument {
	title: string;
	subtitle?: string;
	slug: Slug;
	author?: string;
	mediaUrl?: string;
	category?: string;
	tags?: string[];
	mainImageUrl?: string;
	description?: string;
	body?: Body[];
	publishedAt?: string;
	writtenAt?: string;
	downloadLink?: string;
	downloadMeta?: string;
}
